src/mongo/delegates/Views.class.php

```diff
@@ -396,58 +406,16 @@ public function generateView($viewId,$resource=null,$context=null,$queueName=nul
             $t = new \Tripod\Timer();
             $t->start();
 
-            $from = $this->getFromCollectionForViewSpec($viewSpec);
+            $from = $viewSpec['from'];
             $collection = $this->config->getCollectionForView($this->storeName, $viewId);
 
+            $this->ensureIndices($viewSpec, $collection);
```

Comment:
All good.  I realise there's a bunch of stuff here that I'll hardly be the first to raise, and understand not wanting to optimise things without test coverage.  I'm going for things that I'm changing, and trying not to pull anything I need not change into the mix.